tests/test_mysql_binlog.py

```diff
@@ -0,0 +1,705 @@
+                                    os.getenv('TAP_MYSQL_PASSWORD'),
+                                    os.getenv('TAP_MYSQL_PORT')] if x == None]
+        if len(missing_envs) != 0:
+            #pylint: disable=line-too-long
+            raise Exception("set TAP_MYSQL_HOST, TAP_MYSQL_PORT, TAP_MYSQL_DBNAME, TAP_MYSQL_USER, TAP_MYSQL_PASSWORD")
+
+        connection = self.get_db_connection()
+
+        with connection.cursor() as cur:
+            create_databases_sql = """
+                DROP DATABASE IF EXISTS {};
+                CREATE DATABASE {};
+            """.format(self.database_name(), self.database_name())
+
+            cur.execute(create_databases_sql)
+
+            cur.execute("""
+            SELECT EXISTS (
+            SELECT 1
+            FROM  information_schema.tables
+            WHERE  table_schema = %s
+            AND  table_name =   %s);""",
+                        [self.database_name(), self.table_name()])
+
+            existing_table = cur.fetchone()[0]
+
+            if existing_table:
+                cur.execute("DROP TABLE {}.{}".format(self.database_name(), self.table_name()))
+
+            create_table_sql = """
+CREATE TABLE {}.{} (
+            id                     BIGINT PRIMARY KEY,
+            our_char               CHAR,
+            our_enum               ENUM('one', 'two', 'three'),
+            our_longtext           LONGTEXT,
+            our_mediumtext         MEDIUMTEXT,
+            our_text               TEXT,
+            our_varchar            VARCHAR(255),
+            our_unsigned_tinyint   TINYINT UNSIGNED,
+            our_signed_tinyint     TINYINT,
+            our_unsigned_smallint  SMALLINT UNSIGNED,
+            our_signed_smallint    SMALLINT,
+            our_unsigned_mediumint MEDIUMINT UNSIGNED,
+            our_signed_mediumint   MEDIUMINT,
+            our_unsigned_int       INT UNSIGNED,
+            our_signed_int         INT,
+            our_unsigned_bigint    BIGINT UNSIGNED,
+            our_signed_bigint      BIGINT,
+            our_unsigned_decimal_1 DECIMAL(11,2) UNSIGNED,
+            our_signed_decimal_1   DECIMAL(11,2),
+            our_unsigned_decimal_2 DECIMAL UNSIGNED,
+            our_signed_decimal_2   DECIMAL,
+            our_unsigned_float     FLOAT UNSIGNED,
+            our_signed_float       FLOAT,
+            our_unsigned_double    DOUBLE UNSIGNED,
+            our_signed_double      DOUBLE,
+            our_bit_1              BIT(1),
+            our_datetime           DATETIME,
+            our_timestamp          TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+            our_date               DATE,
+            our_time               TIME,
+            our_boolean            BOOLEAN
+)
+""".format(self.database_name(), self.table_name())
+
+            cur.execute(create_table_sql)
+
+            for record in [rec_1, rec_2]:
+                self.insert_record(cur, record)
+
+
+    def test_run(self):
```

Comment:
In general I feel our test run methods are too long. Is there any way to split this up to make it easier to read?

Reply:
Not sure, there's probably a lot that can be done in a tap-tester library API refactor sort of situation. Like common convenience methods.